crates/astria-sequencer-relayer/src/relayer/celestia_client/tests.rs

```diff
@@ -55,8 +57,10 @@ fn new_msg_pay_for_blobs_should_fail_for_large_blob() {
         commitment: Commitment([0; 32]),
     };
     let error = new_msg_pay_for_blobs(&[blob], Bech32Address("a".to_string())).unwrap_err();
-    // allow: `assert!(matches!(..))` provides poor feedback on failure.
-    #[allow(clippy::manual_assert)]
+    #[expect(
+        clippy::manual_assert,
+        reason = "`assert!(matches!(..))` provides poor feedback on failure"
```

Comment:
Also here and below: we should really have a `assert_error_variant` or something like that.